srml/support/src/dispatch.rs

```diff
@@ -67,38 +67,117 @@ pub trait Parameter: Codec + Clone + Eq {}
-/// construct_runtime! macro) and automatically includes all the modules that are used by the
-/// runtime (alongside with a variant called "system").
+/// ## Priviledged Function Example
 ///
-/// A runtime module is a collection of functions unified by a common problem domain and certain
-/// shared types. The "functions" do not actually return values (see Dispatchable) and are only
-/// used for side effects.
+/// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, Trait, ensure_signed};
+///
+/// decl_module! {
+/// 	pub struct Module<T: Trait> for enum Call where origin: T::Origin {
+///		fn my_function() -> Result {
+///			// Your implementation
+/// 			Ok(())
+/// 		}
+///	}
+/// }
+/// # fn main() {}
+/// ```
 ///
-/// For every module an associated enum (usually "Call") is generated with every variant
-/// corresponding to a function of the module. This enum implements Callable and thus its values
-/// can be used as an extrinsic's payload.
+/// If the `origin` param is omitted, the macro adds it as the first parameter and adds `ensure_root(origin)` as the first line of the function.
 ///
-/// The `on_initialize` and `on_finalize` functions are special, since it can either take no
-/// parameters, or one parameter, which has the runtime's block number type.
+/// ## Multiple Module Instances Example
 ///
-/// ### Module with instances
+/// A Substrate module can be built such that multiple instances of the same module can be used within a single runtime.
+/// For example, the [Balances module](../srml_balances/index.html) can be added multiple times to your runtime
+/// in order to support multiple, independent currencies for your blockchain. Here is an example of how you would declare
+/// such a module using the `decl_module!` macro:
 ///
-/// decl_module! support modules with instances with the following syntax: (DefaultInstance type is
-/// optionnal)
-/// ```nocompile
-/// pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {}
 /// ```
+/// # #[macro_use]
+/// # extern crate srml_support;
+/// # use srml_support::dispatch::Result;
+/// # use srml_system::{self as system, ensure_signed};
+/// # pub struct DefaultInstance;
+/// # pub trait Instance {}
+/// # impl Instance for DefaultInstance {}
+/// 
+/// pub trait Trait<I: Instance=DefaultInstance>: system::Trait {}
+/// 
+/// decl_module! {
+/// 	pub struct Module<T: Trait<I>, I: Instance = DefaultInstance> for enum Call where origin: T::Origin {
+/// 		// Your implementation
+/// 	}
+/// }
+/// # fn main() {}
+/// ```
+///
+/// ## Reserved Functions
+///
+/// The following are reserved function signatures.
+///
+/// * `deposit_event`: Emits [events](https://docs.substrate.dev/docs/event-enum) for the module.
+///
+/// The following reserved functions also take an optional input `n: T::BlockNumber`:
```

Comment:
I don't think we should have the variable `n` here, because you can really call it whatever you want (unless this is required by the macro, @bkchr ?) and we should explicitly say that it is the block number. `BlockNumber` is a type provided by Substrate, but doesn't have to be a block number. 

E.g.: https://github.com/paritytech/substrate/blob/master/srml/session/src/lib.rs#L89

Reply:
The name is up to you and does not need to be `n`.
Regarding the block number, I would always expect that something that uses the BlockNumber type is also a block number. There are maybe exceptions of this rule, but probably not that many.